src/ctapipe/reco/stereo_combination.py

```diff
@@ -404,3 +478,313 @@ def predict_table(self, mono_predictions: Table) -> Table:
+    Especially at low energies, the DISP sign reconstruction can be quite uncertain.
+    To solve this head-tail ambiguity this algorithm does the following for all valid
+    telescopes of an array event:
+
+    1. Two possible FoV positions (lon/lat) are computed from the Hillas centroid,
+       the DISP parameter, and the Hillas orientation angle (Hillas psi).
+
+    2. For every telescope pair, all four SIGN combinations are evaluated.
+       The SIGN pair that minimizes the angular distance between the two predicted
+       positions is selected, optionally weighted by the telescope-wise DISP
+       sign score. A weighted mean for the minimum distance per telescope pair
+       is calculated.
+
+    3. A second weighted mean FoV direction of all telescope-pair minima is computed
+       afterwards and transformed into horizontal (Alt/Az) coordinates.
+
+    See :class:`StereoCombiner` for a description of the general interface.
+
+    Notes
+    -----
+    - Only geometry (:class:`~ctapipe.reco.ReconstructionProperty.GEOMETRY`)
+      is supported.
+    - Weighting options follow the same convention as in :class:`StereoMeanCombiner`
+      (none, intensity, konrad).
+    - The DISP sign-score can optionally be used to prefer SIGN combinations
+      with higher reliability when resolving the DISP head–tail ambiguity.
+    """
+
+    sign_score_limit = Float(
+        default_value=0.85,
+        min=0,
+        max=1.0,
+        allow_none=True,
+        help=(
+            "Lower-limit for the telescope-wise sign scores to consider in the weighting "
+            "of the distances per telescope combination. The value must be between 0 and "
+            "and 1. Telescope events with a sign score above this limit are taken "
+            "preferably into account for calculating the minimum distance per telescope "
+            "combination. Set to None to disable this feature."
+        ),
+    ).tag(config=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.supported = ReconstructionProperty.GEOMETRY
+        if self.property not in self.supported:
+            raise NotImplementedError(
+                f"Combination of {self.property} not implemented in {self.__class__.__name__}"
+            )
+
+        self.n_tel_combinations = 2
+
+    def _calc_dist_weights(self, disp, sign_score, signs):
+        dist_weight = np.ones(2)
+        if self.sign_score_limit is None:
+            return dist_weight
+
+        if sign_score >= self.sign_score_limit:
+            dist_weight[np.sign(disp) == signs] = 1 / (1 + sign_score)
+
+        return dist_weight
+
+    def _combine_altaz(self, event):
+        ids = []
+        fov_lon_values = []
+        fov_lat_values = []
+        weights = []
+        dist_weights = []
+
+        signs = np.array([-1, 1])
+
+        for tel_id, dl2 in event.dl2.tel.items():
+            if dl2.geometry[self.prefix].is_valid:
```

Comment:
Shouldn't this check `disp` and not `geometry`? The combiner is the thing that fills geometry, right?